sktime/split/temporal_train_test_split.py

```diff
@@ -83,46 +80,155 @@ def temporal_train_test_split(
                 "also be specified."
             )
         return _split_by_fh(y, fh, X=X)
+
+    temporal_splitter = TemporalTrainTestSplitter(
+        test_size=test_size, train_size=train_size
+    )
+
+    y_train, y_test = list(temporal_splitter.split_series(y))[0]
+
+    if X is not None:
+        from sktime.split import SameLocSplitter
+
+        X_splitter = SameLocSplitter(temporal_splitter, y)
+        X_train, X_test = list(X_splitter.split_series(X))[0]
+
+        return y_train, y_test, X_train, X_test
     else:
-        pd_format = isinstance(y, pd.Series) or isinstance(y, pd.DataFrame)
-        if pd_format is True and isinstance(y.index, pd.MultiIndex):
-            ys = get_time_index(y)
-            # Get index to group across (only indices other than timepoints index)
-            yi_name = y.index.names
-            yi_grp = yi_name[0:-1]
-
-            # Get split into test and train data for timeindex only
-            series = (ys,)
-            yret = _train_test_split(
-                *series,
-                shuffle=False,
-                stratify=None,
-                test_size=test_size,
-                train_size=train_size,
-            )
+        return y_train, y_test
```

Comment:
This probably does not need an else block, since if block contains a return already.

Reply:
I thought the else block made the condition easier to spot for a reader of the code, because it matches the indentation and can be tracked up to the `if` condition.

If we remove the `else`, the `if` condition will be harder to find.

This is my rationale (in general) to do this even if you have `if`-`return`, but I'm not too strongly opined about this.

Reply:
For instance, I did the opposite with `if fh is not None:`, because the `else` was very long.

Reply:
I made a change to the code flow, and added comments - better? I went with your suggestion, but added comments to the `fh` and the `X` branching to make it clear to the reader

Reply:
It was a non-blocking comment really. Usually tools like `pylint` will complain against these (ref. [R1705](https://pylint.pycqa.org/en/stable/user_guide/messages/refactor/no-else-return.html)).

The change looks good to me. `pylint` or etc. may still complain that function return signature vary between different return statements (ref. [R1710](https://pylint.readthedocs.io/en/stable/user_guide/messages/refactor/inconsistent-return-statements.html)), but I don't think there's a way to solve that here at all.